### PR TITLE
Add dependency functional

### PR DESCRIPTION
**Pull request by RiByryn**
Add implementation of test dependencies without using third party plugins

- Skips the test case if the main test fails/skipped

```python
@pytest.mark.case_dependency("test_a::test_one")
```

-  Skips the test module if the main test fails/skipped

```python
pytestmark = pytest.mark.module_dependency("test_a::test_one")
```

### CODE REVIEW

File: hardpy/hardpy_panel/frontend/src/hardpy_test_view/SuiteList_v3.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export interface TestRunI {
     stop_time?: number,
     progress?: number,
     drivers?: DriversInfo,
-    artifact?: Record<string, unknown>,
+    artifact?: Record<string, unknown>
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply hardpy/hardpy_panel/frontend/src/hardpy_test_view/SuiteList_v3.tsx:49]: returned

File: hardpy/pytest_hardpy/db/const.py [hunk 2]
```diff
@@ -23,3 +23,4 @@ class DatabaseField(str, Enum):  # noqa: WPS600
     TEST_STAND = "test_stand"
     SERIAL_NUMBER = "serial_number"
     DRIVERS = "drivers"
+    DEPENDENCY = "dependency"
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply hardpy/pytest_hardpy/db/const.py:26]: deleted

File: hardpy/pytest_hardpy/db/schema.py [hunk 3]
```diff
@@ -33,6 +33,7 @@ class CaseStateStore(IBaseResult):
     }
     """
 
+    artifact: Optional[dict] = {}
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply hardpy/pytest_hardpy/db/schema.py:36]: returned

File: hardpy/pytest_hardpy/plugin.py [hunk 4]
```diff
@@ -2,6 +2,8 @@
 # GNU General Public License v3.0 (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 import signal
+import pytest
+import re
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply hardpy/pytest_hardpy/plugin.py:6]: changed

File: hardpy/pytest_hardpy/plugin.py [hunk 5]
```diff
@@ -19,7 +21,8 @@
     ExitCode,
 )
 
-from hardpy.pytest_hardpy.reporter import HookReporter
+from hardpy.pytest_hardpy.db import DatabaseField as DF
+from hardpy.pytest_hardpy.reporter import HookReporter, RunnerReporter
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply hardpy/pytest_hardpy/plugin.py:25]: deleted

File: hardpy/pytest_hardpy/reporter/hook_reporter.py [hunk 6]
```diff
@@ -296,3 +295,19 @@ def _get_case_name(self, node_info) -> str:
             str: case name
         """
         return node_info.case_name if node_info.case_name else node_info.case_id
+
+    def _get_dependency_status(self, node_info: NodeInfo) -> str | None:
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply hardpy/pytest_hardpy/reporter/hook_reporter.py:299]: deleted

File: examples/project/minute_parity/test_2.py [hunk 7]
```diff
@@ -4,7 +4,9 @@
 from driver_example import DriverExample
 
 
-pytestmark = pytest.mark.module_name("Main tests")
+pytestmark = [
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply examples/project/minute_parity/test_2.py:7]: added

File: hardpy/pytest_hardpy/plugin.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (xorialexandrov) [reply hardpy/pytest_hardpy/plugin.py]: Pytest is not a built-in python package

File: docs/examples/minute_parity.md [hunk 8]
```diff
@@ -110,15 +111,21 @@ def test_dut_info(module_log: logging.Logger):
     serial_number = str(uuid4())[:6]
     module_log.info(f"DUT serial number {serial_number}")
     hardpy.set_dut_serial_number(serial_number)
-    info = {"batch": "test_batch", "board_rev": "rev_1"}
+    info = {
```
Comment (xorialexandrov) [inline docs/examples/minute_parity.md:114]: In `.md` code must be compact.

File: examples/project/minute_parity/test_1.py [hunk 9]
```diff
@@ -22,6 +22,7 @@ def test_dut_info(module_log: logging.Logger):
 
 
 @pytest.mark.case_name("Test stand info")
+@pytest.mark.case_dependency("test_1::test_dut_info")
```
Comment (xorialexandrov) [inline examples/project/minute_parity/test_1.py:25]: Why do we need to depend on `test_dut_info` if this test is always `True`?

Comment (RiByryn) [reply examples/project/minute_parity/test_1.py:25]: deleted dependency

File: docs/examples/minute_parity.md [hunk 10]
```diff
@@ -95,6 +95,7 @@ Contains tests related to preparation for the testing process:
 - The name of the test cases for the web interface is set to `pytest.mark.case_name`;
 - The device under test (DUT) info is stored in the database in `test_dut_info`;
 - The test stand info is store in the database in `test_stand_info`;
+- Test `test_stand_info` depends on test `test_dut_info`, dependency is set to `pytest.mark.case_dependency`
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/examples/minute_parity.md:98]: added info

File: docs/documentation/pytest_hardpy.md [hunk 11]
```diff
@@ -267,6 +267,26 @@ Sets a text name for the test module (file) (default: module name)
...
 ```
 
+#### case_dependency
+
+Skips the test case if the main test fails/skipped/errored
+
+**Example:**
+
+```python
+@pytest.mark.case_dependency("test_a::test_one")
```
Comment (xorialexandrov) [inline docs/documentation/pytest_hardpy.md:277]: Need more code context.

File: docs/examples/minute_parity.md [hunk 12]
```diff
@@ -110,7 +110,7 @@ def test_dut_info(module_log: logging.Logger):
     serial_number = str(uuid4())[:6]
     module_log.info(f"DUT serial number {serial_number}")
     hardpy.set_dut_serial_number(serial_number)
-    info = {"batch": "test_batch", "board_rev": "rev_1"}
+    info = {"batch": "test_batch", "board_rev": "rev_1",}
```
Comment (xorialexandrov) [inline docs/examples/minute_parity.md:113]: No comma needed.

File: docs/documentation/pytest_hardpy.md (hunk 11 above)
Comment (RiByryn) [reply docs/documentation/pytest_hardpy.md]: ready

File: docs/examples/minute_parity.md (hunk 12 above)
Comment (RiByryn) [reply docs/examples/minute_parity.md]: ready

**Review by xorialexandrov — APPROVED** [review]

**Pull request merged**